03week/mastermind.js

```diff
@@ -28,18 +28,57 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
-  // your code here
+function generateHint(guess, solution) {
+  // hints will consist of two numbers (2-2)
+  // the first number is the amount of correct letters
+  // the second number is the amount of letters in the correct position
+
+  // split both guess and solution into arrays for iteration
+  guess = guess.split('');
+  solution = solution.split('');
+
+  // compare guess[0] to solution[i]
+  // first check that guess[0]] exist in solution array
+  // if guess[0]] is located in solution index
+  //    add one to the corrLetter accumulator
+  // then check if guess[0] is located in solution[0]
+  //    add one to the corrPosition accumulator
+  // repeat loop for all letters in guess array against solution array
+
+  // initialize the variables corrLetter and corrPostions as accumulators
+  // since these variable would possibly change each round let seemed best
+  let corrLetter = 0;
+  let corrPosition = 0;
+  for (let i = 0; i < solution.length; i++) {
+    if (guess.indexOf(solution[i]) > -1) {
+      corrLetter++;
+    }
+    if (guess[i] === solution[i]){
+      corrPosition++;
+    }
+  }
+
+  return corrLetter + '-' + corrPosition;
 }
 
 function mastermind(guess) {
-  solution = 'abcd'; // Comment this out to generate a random solution
-  // your code here
-}
+  //solution = 'abcd'; // Comment this out to generate a random solution
 
+  // check for correct solution
+  if (guess === solution) {
+    console.log(`You guessed correct!`);
+    process.exit();
+  }
+
+  // print out number of corrLetter/corrPosition (corrLetter - corrPosition)
+  // since hint would change each round let seemed the best solution
+  let hint = generateHint(guess, solution);
```

Comment:
Also, no need to pass solution to the generateHint function, it's a global variable